samples: ble_nus: detect UART line terminator with compare-match (#817)
nrfconnect/sdk-nrf-bm: doc/nrf-bm/release_notes/release_notes_changelog.rst

   * Fixed a bug with the UARTE RX buffer address provided with index 1.

>    * Updated UART reception to detect the line terminator in hardware using the UARTE compare-match filter, reporting each line as its own transfer instead of each received byte.

Reviewer: 'Updated' before 'Fixed'

>    * Updated UART reception to detect the line terminator in hardware using the UARTE compare-match filter, reporting each line as its own transfer instead of each received byte.

Reviewer: ```suggestion
   * Updated UART reception to detect the line terminator in hardware using the UARTE compare-match filter, so that each line is reported as a single transfer instead of one transfer per received byte.
```
same below

No changes since the latest nRF Connect SDK Bare Metal release.

> Peripheral samples

Reviewer: This duplicates the Peripheral samples section (already in line 173)
